### PR TITLE
Fix Run IoTDB main meets 'Wrong max_degree_of_index_node, please set as'

### PR DESCRIPTION
**Pull request by yuqi1129**
## Description

please see https://github.com/apache/iotdb/issues/2900

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

##### Key changed/added classes (or packages if there are too many classes) in this PR

Just change a config

### CODE REVIEW

File: tsfile/src/main/java/org/apache/iotdb/tsfile/common/conf/TSFileConfig.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
   /** The maximum number of data points in a page, default value is 1024 * 1024. */
   private int maxNumberOfPointsInPage = 1024 * 1024;
   /** The maximum degree of a metadataIndex node, default value is 1024 */
-  private int maxDegreeOfIndexNode = 256;
+  private int maxDegreeOfIndexNode = 1024;
```
Comment (wangchao316) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/common/conf/TSFileConfig.java:78]: hi , maxDegreeOfIndexNode should be Consistent as config item.  you should alter this config too. 
https://github.com/apache/iotdb/blob/4f4ee906c6a10915d86ce900ede7174ab161fbfa/server/src/assembly/resources/conf/iotdb-engine.properties#L570

**Review by wangchao316 — CHANGES_REQUESTED** [review]

File: tsfile/src/main/java/org/apache/iotdb/tsfile/common/conf/TSFileConfig.java (hunk 1 above)
Comment (yuqi1129) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/common/conf/TSFileConfig.java:78]: OK, Thanks for your careful review~

**Review by wangchao316 — APPROVED** [review]
LGTM

**Comment by JackieTien97** [comment]
The change of this default config value aims to adapt to the new Tsfile format, because the new tsfile format has moved the chunk metadata list next to the timeseries metadata which locates in the leaf node of metadata tree, so we reduce the default number of maxDegreeOfIndexNode.
If you occur the error, it means that you use the master version which is not merged with NewTsfileV2 to generate the tsfile, and then use the current master version to restart with the stale data.

**Pull request closed without merging**